File: framework/test2/char/chars1d/Chars1d.scala
import soda.scala.unittest._

import scala.collection.mutable
import scala.reflect.runtime.universe.typeOf

object Solution {
  def doubleList(chars: Array[Char]): List[Char] = {
    var res = mutable.ArrayBuffer[Char]()
    for (ch <- chars) {
      res += ch
    }
    for (ch <- chars) {
      res += ch
    }
    return res.toList
  }
}

class Chars1d extends (String => String) {
  override def apply(text: String): String = {
    val work = GenericTestWork.create1(Solution.doubleList)
    work.compareSerial = true
    work.run(text)
  }
}

object Chars1d extends App {
  println(new Chars1d()(Utils.fromStdin()))
}

File: framework/test2/char/chars1d/Chars1d.cs
using Soda.Unittest;

public class Solution {
    public IList<char> DoubleList(char[] chars)
    {
        var res = new List<char>();
        foreach (char ch in chars) {
            res.Add(ch);
        }
        foreach (char ch in chars) {
            res.Add(ch);
        }
        return res;
    }
}

public class Chars1d
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().DoubleList));
        // work.SetValidator((e, r) => ...);
        work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.readStdin()));
    }
}
